Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6174,6 +6174,16 @@ void WebPageProxy::updateRemoteFrameSize(WebCore::FrameIdentifier frameID, WebCo
     sendToProcessContainingFrame(frameID, Messages::WebPage::UpdateFrameSize(frameID, size));
 }
 
+void WebPageProxy::updateSandboxFlags(IPC::Connection& connection, WebCore::FrameIdentifier frameID, WebCore::SandboxFlags sandboxFlags)
+{
+    if (RefPtr frame = WebFrameProxy::webFrame(frameID)) {
+        RefPtr process = dynamicDowncast<WebProcessProxy>(AuxiliaryProcessProxy::fromConnection(connection));
+        RefPtr parentFrame = frame->parentFrame();
+        MESSAGE_CHECK(process, parentFrame && &parentFrame->process() == process.get());
```

Comment:
Don’t need the hash map lookup:
```suggestion
        RefPtr parentFrame = frame->parentFrame();
        MESSAGE_CHECK(process, parentFrame && parentFrame->process().hasConnection(connection));
```

Reply:
hasConnection could be used, but we do have to do the hash lookup to determine which process to apply the message check to, so I'm not sure it would improve anything.

Reply:
you could use `MESSAGE_CHECK_BASE(connection, parentFrame && parentFrame->process().hasConnection(connection));`